src/prommis/uky/costing/tests/test_ree_costing.py

```diff
@@ -1109,6 +1123,19 @@ def test_REE_costing(self, model):
         # check that the model is set up properly and has 0 degrees of freedom
         assert degrees_of_freedom(model) == 0
 
+    @pytest.mark.component
+    def test_full_model_diagnostics(self, model):
+        dt = DiagnosticsToolbox(model)
+
+        try:
+            dt.assert_no_structural_warnings()
+        except AssertionError as e:
```

Comment:
I would generally leave this out of the test - you have no idea exactly how the test might fail in the future so it is hard to predict what methods you might need to call for debugging. At most I would have `report_structural_issues` here as you don;t know what other methods might be useful.